internal/command/org_metadata_model.go

```diff
@@ -0,0 +1,94 @@
+package command
+
+import (
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/repository/org"
+)
+
+type OrgMetadataWriteModel struct {
+	MetadataWriteModel
+}
+
+func NewOrgMetadataWriteModel(orgID, resourceOwner, key string) *OrgMetadataWriteModel {
+	return &OrgMetadataWriteModel{
+		MetadataWriteModel{
+			WriteModel: eventstore.WriteModel{
+				AggregateID:   orgID,
+				ResourceOwner: resourceOwner,
+			},
+			Key: key,
+		},
+	}
+}
+
+func (wm *OrgMetadataWriteModel) AppendEvents(events ...eventstore.Event) {
+	for _, event := range events {
+		switch e := event.(type) {
+		case *org.MetadataSetEvent:
+			wm.MetadataWriteModel.AppendEvents(&e.SetEvent)
+		case *org.MetadataRemovedEvent:
+			wm.MetadataWriteModel.AppendEvents(&e.RemovedEvent)
+		case *org.MetadataRemovedAllEvent:
+			wm.MetadataWriteModel.AppendEvents(&e.RemovedAllEvent)
+		}
+	}
+}
+
+func (wm *OrgMetadataWriteModel) Query() *eventstore.SearchQueryBuilder {
+	return eventstore.NewSearchQueryBuilder(eventstore.ColumnsEvent).
+		ResourceOwner(wm.ResourceOwner).
+		AddQuery().
+		AggregateIDs(wm.MetadataWriteModel.AggregateID).
+		AggregateTypes(org.AggregateType).
+		EventTypes(
+			org.MetadataSetType,
+			org.MetadataRemovedType,
+			org.MetadataRemovedAllType).
+		Builder()
+}
+
+type OrgMetadataListWriteModel struct {
+	MetadataListWriteModel
+}
+
+func NewOrgMetadataListWriteModel(orgID, resourceOwner string) *OrgMetadataListWriteModel {
+	return &OrgMetadataListWriteModel{
+		MetadataListWriteModel{
+			WriteModel: eventstore.WriteModel{
+				AggregateID:   orgID,
+				ResourceOwner: resourceOwner,
+			},
+			metadataList: make(map[string][]byte),
+		},
+	}
+}
```

Comment:
I think orgID and resourceowner is redundant

```suggestion
func NewOrgMetadataListWriteModel(orgID string) *OrgMetadataListWriteModel {
	return &OrgMetadataListWriteModel{
		MetadataListWriteModel{
			WriteModel: eventstore.WriteModel{
				AggregateID:   orgID,
				ResourceOwner: orgID,
			},
			metadataList: make(map[string][]byte),
		},
	}
}
```